contracts/cw1-subkeys/src/contract.rs

```diff
@@ -73,17 +84,25 @@ where
         res.log = vec![log("action", "execute"), log("owner", env.message.sender)];
         Ok(res)
     } else {
-        let mut allowances = allowances(&mut deps.storage);
-        let allow = allowances.may_load(owner_raw.as_slice())?;
-        let mut allowance =
-            allow.ok_or_else(|| StdError::not_found("No allowance for this account"))?;
         for msg in &msgs {
             match msg {
+                CosmosMsg::Staking(staking_msg) => {
+                    let permissions = permissions(&mut deps.storage);
```

Comment:
This is fine and explicit. I generally make this more compact.

```rust
let perm = permissions(&mut deps.storage).load(owner_raw.as_slice())?;
check_staking_permissions(staking_msg, perm)?;
```

But it is fine as is. (and with custom error message)